.pre-commit-config.yaml

```diff
@@ -0,0 +1,10 @@
+#largely copied from https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v2.4.0
+      hooks:
+          - id: flake8
+    - repo: https://github.com/psf/black
+      rev: stable
+      hooks:
+          - id: black
```

Comment:
[This article](https://mitelman.engineering/blog/python-best-practice/automating-python-best-practices-for-a-new-project/) also recommends the following config file for Black (you put it in a file called `pyproject.toml`): 

```
[tool.black]
line-length = 79
target-version = ['py38']
include = '\.pyi?$'
exclude = '''

(
  /(
      \.eggs         # exclude a few common directories in the
    | \.git          # root of the project
    | \.hg
    | \.mypy_cache
    | \.tox
    | \.venv
    | _build
    | buck-out
    | build
    | dist
  )/
  | foo.py           # also separately exclude a file named foo.py in
                     # the root of the project
```
But as they say, really the most important line is `line-length`, it seems to work fine without the rest. (And the reason we want to use 79 is because of PEP-8)